sympy/geometry/ellipse.py

```diff
@@ -718,13 +718,19 @@ def is_tangent(self, o):
         Examples
         ========
 
-        >>> from sympy import Point, Ellipse, Line
+        >>> from sympy import Point, Ellipse, Line, Circle, Polygon
         >>> p0, p1, p2 = Point(0, 0), Point(3, 0), Point(3, 3)
         >>> e1 = Ellipse(p0, 3, 2)
         >>> l1 = Line(p1, p2)
         >>> e1.is_tangent(l1)
         True
-
+        >>> p3 = Point(0,3)
+        >>> c = Circle(Point(0,0),3)
+        >>> poly = Polygon(p0,p1,p2,p3)
+        >>> c.is_tangent(poly)
+        False
+        >>> c.is_tangent(poly,True)
```

Comment:
Keyword arguments like this should be spelled out explicitly.

Reply:
Instead of a keyword, another method like `a.has_tangent(b)` could be created.

See also comment [here](https://github.com/sympy/sympy/pull/13496#issuecomment-344555246)

Reply:
We could also use ternary logic: True = all sides, None = at least one side, False = no sides